hooks/pre/35-change-evr-extension-ownership.rb

```diff
@@ -0,0 +1,28 @@
+# In Katello 4.16, the 'evr' extension is removed from PostgreSQL and integrated into the Katello database via a migration.
+# This hook ensures the 'evr' extension's ownership is transferred to the 'foreman' user so migrations can act on it.
+
+if (local_foreman_db? || devel_scenario?) && execute("rpm -q postgresql-evr", false, false)
+  if app_value(:noop)
+    logger.debug("Would start postgresql service")
+  else
+    start_services(['postgresql'])
+  end
+
+  database = param_value('foreman', 'db_database') || 'foreman'
+  username = param_value('foreman', 'db_username') || 'foreman'
+
+  if local_db_exists?(database)
+    sql = "UPDATE pg_extension SET extowner = (SELECT oid FROM pg_authid WHERE rolname='#{username}') WHERE extname='evr'"
+    if app_value(:noop)
+      logger.debug("Would execute the following SQL statement to update ownership of the evr extension: #{sql}")
+    else
+      logger.debug("Updating ownership of the evr extension")
+      _, success = execute_preformatted_as('postgres', "psql -d #{database} -c \"#{sql}\"", false, true)
+      unless success
+        logger.notice("Failed to update ownership of the evr extension.")
```

Comment:
Warning or even error that halts execution?

Reply:
Maybe. It was a warning before, so that's what caught my eye.

What happens if it is not updated? Later the removal of the extension fails?

Reply:
Ya..The evr extension is dropped in katello migration here https://github.com/Katello/katello/pull/11159/ which assumes the foreman user is the owner of the extension. We could fail here instead of later on during db:migrate.

Reply:
Updated to fail and exit:
```
fail_and_exit("Failed to update ownership of the evr extension. Please make sure the following sql succeeds before proceeding: #{sql}")
```

Reply:
@evgeni @ekohl Is this what y'all had in mind?